cpp/data_structures/dynamic_tree/dynamic_tree.hpp

```diff
@@ -289,12 +290,12 @@ namespace cpstl {
 
             // current_node => it's a node represent you position in tree
             // "important !" for many operation like search , movement , ...
-            DT_Node<v> *current_node;
+            DTNode<v> *current_node;
 
             // constructor
             DynamicTree(std::string root_name , v root_value) {
                 // parent of root must be null
-                root = DT_Node<v>( root_name , root_value , NULL ); 
+                root = DTNode<v>( root_name , root_value , NULL ); 
```

Comment:
```suggestion
                root = std::make_shared<DTNode<V>>( root_name , root_value , NULL ); 
```

Reply:
mmm it's look like , i need to studying smart pointers